assets/app/scripts/controllers/create/createFromImage.js

```diff
@@ -83,11 +83,18 @@ angular.module("openshiftConsole")
                     scope.deploymentConfig.envVars[pair[0]] = pair[1];
                   });
 
-                  scope.routing.portOptions = ApplicationGenerator.parsePorts(imageStreamTag.image);
-                  if (scope.routing.portOptions.length){
-                    scope.routing.targetPort = scope.routing.portOptions[0];
-                  } else {
+                  var ports = ApplicationGenerator.parsePorts(imageStreamTag.image);
+                  if (ports.length === 0) {
                     scope.routing.include = false;
+                    scope.routing.portOptions = [];
+                  } else {
+                    scope.routing.portOptions = _.map(ports, function(portSpec) {
+                      return {
+                        port: portSpec.containerPort,
+                        label: portSpec.containerPort + "/" + portSpec.protocol
+                      };
+                    });
+                    scope.routing.targetPort = scope.routing.portOptions[0];
```

Comment:
don't we want this to use the name we generated for the service, not the number?